skytemple_files/common/ppmdu_config/pmdsky_debug/test/migration_test.py

```diff
@@ -0,0 +1,243 @@
+                                 'AppraiserMenuConfirm': (True, False),
+                                 'AppraiserSubMenu': (True, False)},
+    'overlay/overlay_0026.bin': {},
+    'overlay/overlay_0027.bin': {'DiscardItemsMainMenu': (True, False),
+                                 'DiscardItemsMenuConfirm': (True, False),
+                                 'DiscardItemsSubMenu1': (True, False),
+                                 'DiscardItemsSubMenu2': (True, False)},
+    'overlay/overlay_0028.bin': {},
+    'overlay/overlay_0029.bin': {'BellyLostTurn': (True, False),
+                                 'BellyLostWtw': (True, False),
+                                 'BellyLostWtw1000': (True, False),
+                                 'EntitySpawnTable': (True, False),
+                                 'FixedFloorOverrides': (True, False),
+                                 'ItemSpawnTable': (True, False),
+                                 'MonsterSpawnTable': (True, False),
+                                 'NectarIqGain': (True, False),
+                                 'TileSpawnTable': (True, False)},
+    'overlay/overlay_0030.bin': {},
+    'overlay/overlay_0031.bin': {'DungeonMainMenu': (True, False),
+                                 'DungeonSubMenu1': (True, False),
+                                 'DungeonSubMenu2': (True, False),
+                                 'DungeonSubMenu3': (True, False),
+                                 'DungeonSubMenu4': (True, False),
+                                 'DungeonSubMenu5': (True, False),
+                                 'DungeonSubMenu6': (True, False)},
+    'overlay/overlay_0032.bin': {},
+    'overlay/overlay_0033.bin': {},
+    'overlay/overlay_0034.bin': {'DungeonDebugMenu': (True, False),
+                                 'UnknownMenuConfirm': (True, False)},
+    'overlay/overlay_0035.bin': {},
+    'overlay/overlay_0036.bin': {}}
+
+
+class MigrationTestCase(unittest.TestCase):
+    """
+    This tests that all blocks and functions that were formely supported by pmd2data.xml are still accessible
+    after migrating to the new pmdsky-debug based solution.
+    """
+    @parameterized.expand((GAME_REGION_EU, GAME_REGION_US))
+    def test_backwards_compatibility(self, region):
+        to_check = {x.filepath: x for x in load_binaries(f"{GAME_VERSION_EOS}_{region}")}
+
+        for bin_path, symbol_specs in OLD_ENTRIES.items():
+            self.assertIn(bin_path, to_check)
+            bin_symbols = to_check[bin_path].symbols
+            for symbol_name, (has_end, is_function) in symbol_specs.items():
+                self.assertIn(symbol_name, bin_symbols, f"Symbol {symbol_name} in binary {bin_path} must be defined.")
+                symbol = bin_symbols[symbol_name]
+                if has_end:
+                    try:
+                        # noinspection PyStatementEffect,PyUnusedLocal
+                        _x = symbol.end
```

Comment:
I think I might've inserted padding bytes on a few more data symbols than the one we found with `IQ_SKILL_RESTRICTIONS`. Maybe we change `has_end: bool` to `end: Optional[int]` and explicitly check the lengths here to make sure there are no other surprises.

Reply:
I didn't want these tests to include the entire old lookup values, I did write a small test script to check, which can be found here: 
https://gist.github.com/theCapypara/4293b80d1251c0a3b9627528b9d033df

I need to re-check the `SKIP`ed values before merging again, but I think none of these rely on the "correct" end being specified at the moment so it's fine.